mhkit/wave/io/wave_hindcast.py

```diff
@@ -0,0 +1,84 @@
+import pandas as pd
+import numpy as np
+from rex import WaveX, MultiYearWaveX
+
+def read_US_wave_dataset(wave_path, parameter, lat_lon, years=None, tree=None, 
+                                 unscale=True, str_decode=True, hsds=True):
+    
+        """
+        Reads data from the WPTO wave hindcast data hosted on AWS. 
+
+        Note: To access the WPTO hindcast data, you will need to configure h5pyd for data access on HSDS. 
+        To get your own API key, visit https://developer.nrel.gov/signup/. 
+
+        To configure h5phd type 
+        hsconfigure
+        and enter at the prompt:
+        hs_endpoint = https://developer.nrel.gov/api/hsds
+        hs_username = None
+        hs_password = None
+        hs_api_key = {your key}
+        Parameters
+        ----------
+        wave_path : string
+            Path to US_Wave .h5 files
+            Available formats:
+                /nrel/US_wave/US_wave$_{year}.h5
+                /nrel/US_wave/US_wave$_*.h5 (Only use when specifying years parameter)
```

Comment:
Not clear to me what I would do here. 

- It looks to me like {year} and * are both used as wildcards?
- I think these should be in quotes since they are a string (e.g. f'/nrel/US_wave/US_wave$_{year}.h5')?  
- I think an example with a description included with the "Path to US_Wave .h5 files" would help.

Reply:
Using {year} is primarily for when you are requesting a single year, or a whole decade, like 199*. The simple * combined with the years parameter would be if you want a few years, I.e. 2005, 2006,2007. Some of this stems from how the rex package works. What we can do is do away with the "wave_path" parameter all together and have a dataset type parameter with options like "spatial", "virtual_bouy", and others as they become available. "Virtual_buoy" data is not quite ready yet but is expected to be very soon. We can then make the years parameter required. 

I like this option because then users do not need to mess around with paths to the data.

Reply:
If I make my proposed change right now there will only be one option for "data_type". @ssolson do you think that will be weird? Like I said, at least one more option will be added soon so whatever route I take I would like to minimize having to change too much once there are more options.

Reply:
I do not think it is weird. I still think the `wave_path` parameter is unclear. I don't think I should be specifying a path. I believe I should be specifying an option. If the difference in the path is one year vs multi years then the function should make that switch for me while I simply specify either 1 year or multiple years. Again take this suggestion with a grain of salt as I am not grasping the purpose of this.

Reply:
OK, I will work on my proposed change. Hopefully afterwards how to use the function will be more clear. There will then no longer be a need to specify a path at all.